### DISCUSSION TITLE
Kinematics

### DISCUSSION THREAD

**Discussion by dglazier** [dglazier/chanser#22] [Development]
The kinematics class can be extended to include additional useful/ common calculations. Please make some suggestions here for what should be included. Also feel free to volunteer for coding these ( a good exersize that will aid understanding of programming and reaction kinematics).

**Comment by dglazier** [comment]
From the problems discussion we should add basics LorentzVector operations as given in GenVector helper functions.

**Comment by naidoo88** [comment]
Shortly I will need to implement a function (which I use elsewhere) to calculate Trento-Phi (the angle between leptonic and  hadronic planes) for a given final state, as well as the coplanarity angle between the possible definitions of planes.

I think it would be generally useful for anyone performing deep-process analyses, so I'm happy to help include this if you would like.

**Reply by dglazier** [reply]
Sounds good. When you are ready to add code, please try and follow the gitflow model posted earlier.

**Comment by rwishart1** [comment]
Currently I am using a function to calculate t' = t - tmin, this might be useful in the class. I'd be happy to code this up at some time soon

**Reply by dglazier** [reply]
Thanks Robert, in principle there is a function in Kinematics to do this (at least t and t0 seperately)
https://github.com/dglazier/chanser/blob/master/core/Kinematics.h#L172
But it is marked PROBALY NOT CORRECT, so it would be good if you could add definately correct functions to replace these.

**Reply by rwishart1** [reply]
Ok that sounds good, I will take a look at this.

**Reply by dglazier** [reply]
To gt back to this, Robert made a kine branch with the t0 correction. I modified this to pass const references rather than objects as it should be a bit faster.
Please use this branch for pushing any further changes to Kinematics.h
